packages/polling-controller/src/PollingController.ts

```diff
@@ -49,19 +54,26 @@ function PollingControllerMixin<TBase extends Constructor>(Base: TBase) {
      * Starts polling for a networkClientId
      *
      * @param networkClientId - The networkClientId to start polling for
+     * @param options - The options used to group the polling events
      * @returns void
      */
-    startPollingByNetworkClientId(networkClientId: NetworkClientId) {
+    startPollingByNetworkClientId(
+      networkClientId: NetworkClientId,
+      options: object = {},
+    ) {
       const innerPollToken = random();
-      if (this.#networkClientIdTokensMap.has(networkClientId)) {
-        const set = this.#networkClientIdTokensMap.get(networkClientId);
+
+      const id = getKey(networkClientId, options);
+
+      if (this.#networkClientIdTokensMap.has(id)) {
+        const set = this.#networkClientIdTokensMap.get(id);
         set?.add(innerPollToken);
```

Comment:
Again another thing for another PR: Should this throw if the set can't be found? Seems like we wouldn't want this method to do nothing.

Reply:
yea i mean ideally typescript should see the `this.#networkClientIdTokensMap.has(id)` and know its a guard